src/Session.js

```diff
@@ -13,12 +13,15 @@ export default class Session {
             this.loginFunction = async () => this._client.loginWithChallengeResponse((d) => account.sign(d).signature, account.address)
         } else if (this.options.provider) {
             const w3 = new Web3(this.options.provider)
-            const accounts = w3.eth.getAccounts()
-            const address = accounts[0]
-            if (!address) {
-                throw new Error('Cannot access account from provider')
+            this.loginFunction = async () => {
+                const address = await w3.eth.getAccounts().then((accounts) => {
```

Comment:
Unusual mix of `await` and `.then()` here! Pick one and stick with it? With `await` it would simplify to something like this (untested):
```
const accounts = await w3.eth.getAccounts()
if (!accounts[0]) {
    throw new Error('Cannot access account from provider')
}
return this._client.loginWithChallengeResponse((d) => w3.eth.personal.sign(d, accounts[0]), accounts[0])
```